main.cpp

```diff
@@ -98,13 +131,29 @@ int main (int argc, char** argv)
             return 0;
         }
 	}
+	
+	self_pipe = createSelfPipe();
+	if( self_pipe.first == -1 || self_pipe.second == -1 )
+	{
+		perror("Error creating pipe file descriptors\n");
+		return 0;
+	}
 
 	if ((!tracemode) && (!DEBUG)){
 		init_ui();
 	}
-
+	
 	if (NEEDROOT && (geteuid() != 0))
 		forceExit(false, "You need to be root to run NetHogs!");
+	
+	fd_set pc_loop_fd_set;
+	std::vector<int> pc_loop_fd_list;
+	bool pc_loop_use_select = true;	
+	
+	if( pc_loop_use_select )
+	{
+		pc_loop_fd_list.push_back(self_pipe.first);
```

Comment:
Is this supposed to be here? It's also present further down, where it makes more sense :)